it/src/main/java/org/corfudb/universe/universe/docker/CorfuServerDockerized.java

```diff
@@ -0,0 +1,312 @@
+                .image(IMAGE_NAME)
+                .hostname(params.getName())
+                .exposedPorts(ports)
+                .cmd("sh", "-c", cmdLine)
+                .build();
+    }
+
+    private void addShutdownHook() {
+        // Just in case a test failed and didn't kill the container
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                kill();
+            } catch (Exception e) {
+                log.debug("Corfu server shutdown hook. Can't kill container: {}", params.getName());
+            }
+        }));
+    }
+
+    private static String getAppVersion() {
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        Model model;
+        try {
+            model = reader.read(new FileReader("pom.xml"));
+            return model.getParent().getVersion();
+        } catch (IOException | XmlPullParserException e) {
+            throw new NodeException("Can't parse application version", e);
+        }
+    }
+
+    /**
+     * This method create a command line string for starting Corfu server
+     *
+     * @return command line parameters
+     */
+    private String getCommandLineParams() {
+        StringBuilder cmd = new StringBuilder();
+        cmd.append("-a ").append(ALL_NETWORK_INTERFACES);
+
+        switch (params.getPersistence()) {
+            case DISK:
+                if (StringUtils.isEmpty(params.getLogDir())) {
+                    throw new UniverseException("Invalid log dir in disk persistence mode");
+                }
+                cmd.append(" -l ").append(params.getLogDir());
+                break;
+            case MEMORY:
+                cmd.append(" -m");
+                break;
+        }
+
+        if (params.getMode() == Mode.SINGLE) {
+            cmd.append(" -s");
+        }
+
+        cmd.append(" -d ").append(params.getLogLevel().toString()).append(" ");
+
+        cmd.append(params.getPort());
+
+        String cmdLineParams = cmd.toString();
+        log.trace("Command line parameters: {}", cmdLineParams);
+
+        return cmdLineParams;
+    }
+
+    public class CorfuManagementServer {
+        private final CorfuRuntime runtime;
+
+        public CorfuManagementServer(ServerParams params) {
+            NodeLocator node = NodeLocator
+                    .builder()
+                    .protocol(NodeLocator.Protocol.TCP)
+                    .host(params.getName())
+                    .port(params.getPort())
+                    .build();
+
+            CorfuRuntimeParameters runtimeParams = CorfuRuntimeParameters
+                    .builder()
+                    .layoutServers(Collections.singletonList(node))
+                    .build();
+
+            runtime = fromParameters(runtimeParams);
+        }
+
+        public boolean add(CorfuServer server) {
+            log.debug("Add node: {}", server.getParams());
+
+            //FIXME fix it corfu runtime and remove this code then
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)